### PR TITLE
Editor: add a classic block guide for users in deprecation group

### LINKED ISSUE #43802: Editor: Contextual help for classic block (after editor deprecation)
> As a user with a preference set for the WordPress.com/Calypso editor,
> I want to use the classic block to edit my content
> So that I can use an editor that is more familiar to me.

## Requirements

Given a user who

- Is in the editor deprecation group (`currentUser.inEditorDeprecationGroup`)
- Has an editor preference of `classic` for the site they are currently editing
- Is creating a new post for the first time after the editor deprecation

Then

1. Automatically insert a classic block into the post that is being created.
1. Show a custom welcome guide that introduces the classic block and instructs how to insert it in the future.
1. If the block editor Welcome Guide would normally show, hide it and show the classic block modal instead. Toggle the welcome guide off, so it isn't shown the next time the user loads the editor.

Design TBD

## Notes

This will load in the iframed block editor in WordPress.com only, so code will live in the wpcom-block-editor Calypso app.

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request

* Add a custom editor guide for users in editor deprecation group to explain how to add a classic block

#### Testing instructions
- Set a user attribute for your user using wpsh on a sandbox $ update_user_attribute( USER_ID, 'editor_deprecation_group', true )
- Open the block editor a site and in Chrome dev tools select `post-new.php` as the javascript context and in console run `wp.data.dispatch( 'core/edit-post' ).toggleFeature( 'welcomeGuide' );` - WPcom guide should load - don't dismiss it
- Now check out this PR and build the wpcom block editor files with yarn build:prod in the apps/wpcom-block-editor/package.json folder.
- Sync the files created by the build in the '/dist' folder with your sandbox widgets/wpcom-block-editor folder
- Also apply D45786-code to sandbox
- Make sure site used above is sandboxed Load editor editor again and make sure the Classic block guide loads instead of the standard WPcom guide
- Click the `Get Started` button to dismiss the dialog then reload page and make sure it does not display again (LocalStorage key for `classic_block_guide_{siteID}_is_dismissed` can be cleared to retest
- Also retest with a user that does not have 'editor_deprecation_group' and make sure the guide does not display

Fixes #43802

### CODE REVIEW

File: apps/wpcom-block-editor/src/wpcom/features/use-classic-block-guide.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import url from 'url';
+/* eslint-enable import/no-extraneous-dependencies */
+
+const parsedEditorUrl = url.parse( window.location.href, true );
+
+const ClassicGuide = () => {
+	const [ isOpen, setOpen ] = useState( true );
+
+	const closeGuide = () => setOpen( false );
+
```
Comment (glendaviesnz) [inline apps/wpcom-block-editor/src/wpcom/features/use-classic-block-guide.js:19]: just added to test if we can use useDispatch to toggle the standard welcome guide in this context - seems to work

**Comment by scruffian** [comment]
I started implementing the design. There are some differences between the mockup and the way the Gutenberg component works:

Screenshot 2020-07-01 at 16 23 15

The main differences are:
- The Gutenberg component has a header with an X to close and dots for the pages
- The "close to finish" button is at the bottom right of the modal
- The modal is forced to be 80% of the viewport height

We have a couple of choices. Either we override a lot of the default setting with custom CSS, which is liable to be fragile, if the underlying CSS for the modal changes. Alternatively we update the design to match the Gutenberg modal styles.

cc @olesyabrk @lessbloat

**Comment by creativecoder** [comment]
> Alternatively we update the design to match the Gutenberg modal styles.

We should update the design--this will be a lot easier to maintain if it closely follows existing Gutenberg conventions.

**Comment by scruffian** [comment]
We also will need this patch D45786-code

**Comment by mtias** [comment]
It doesn't make sense to me to create an entirely new design for a welcome-dialog given core already ships with one. What's the rationale there?

**Comment by mtias** [comment]
Core's welcome guide:

image

**Comment by creativecoder** [comment]
> It doesn't make sense to me to create an entirely new design for a welcome-dialog given core already ships with one.

@mtias Completely agree. When I said "we should update the design", I meant change the design @davemart-in had mocked up for #43802 and instead use something that closely matches what's already in Gutenberg.

**Comment by creativecoder** [comment]
Update on the requirements for this: https://github.com/Automattic/wp-calypso/issues/43802#issuecomment-653242385

**Comment by scruffian** [comment]
@olesyabrk @davemart-in I updated this with the latest designs:

welcome

I added the block inserter image to the copy on the second screen. The gif above shows the core welcome images, which use animated gifs which is pretty nice. If we don't want to do that please can you attach images to this PR or the issue, so I can add them? Thanks!

**Comment by glendaviesnz** [comment]
@scruffian this is looking good. I have removed the hard coded query param and switched it to use the deprecation group flag, so you can retest it using a user in the deprecation group. I have also removed the unused image from the associated diff

**Comment by davemart-in** [comment]
Looking good @scruffian. If we could change the title on the first screen to "Meet the Classic block" so that it fits on one line, also there is some updated designs and copy for the second screen here: https://github.com/Automattic/wp-calypso/issues/43802#issuecomment-655636288 After that I think we'll be all set.

**Comment by scruffian** [comment]
Thanks @davemart-in @olesyabrk, can you provide the image assets too please?

**Comment by olesyabrk** [comment]
Here are the image files:

[Slide one](https://drive.google.com/file/d/1jbOhcv22U0S6v9a-NKFeR6bdH_gY-PoC/view?usp=sharing)
[Slide two](https://drive.google.com/file/d/1hUPOrFBu5h0mf6XegqWfmfnbpaW4tI3r/view?usp=sharing)

There is an [the existing core gif](https://user-images.githubusercontent.com/25157638/81209715-be0a7480-8f85-11ea-81a9-a9005d6c423e.gif) for the second image, but I will need to figure out how to make a gif of the first one if we decide to go that direction.

**Comment by glendaviesnz** [comment]
@olesyabrk - because of the way the core guide is set up the svg images need to include the inner padding, eg. the classic block image currently fills the full image container:

Screen Shot 2020-07-13 at 10 33 51 AM

The insert image one works fine though:

Screen Shot 2020-07-13 at 10 34 09 AM

Are you able to provide the classic block one again with some inner padding please?

**Comment by glendaviesnz** [comment]
> If we could change the title on the first screen to "Meet the Classic block"

Done.

**Comment by aaronrobertshaw** [comment]
This works as advertised for me.

Once the padding is added to the image it should be ready to go.

**Comment by scruffian** [comment]
Let's merge this and update the image in a different PR

**Comment by olesyabrk** [comment]
[Here you go, with padding](https://drive.google.com/file/d/1c_dionFbNNmysKdjLZbrjfKIz--3rSXg/view?usp=sharing), lmk if this works @glendaviesnz !

**Comment by glendaviesnz** [comment]
Thanks @olesyabrk - the padding on the new image is great, but the background color was not quite right compared to the second panel - I assume the background color is supposed to be the same on both so I changed the background to #00a0d2 so it matches the second one, as this the color  set by the core welcome guide - get back to me if there was a reason the image used a different blue.

**Comment by glendaviesnz** [comment]
@olesyabrk - fyi PR with new image at https://github.com/Automattic/wp-calypso/pull/44122

**Comment by olesyabrk** [comment]
@glendaviesnz I'm referencing the new blue for dotcom found on [here](https://www.figma.com/file/DrHwefPlRR0UXbXkRGlH5l/Welcome-Tour?node-id=619%3A0) #1381D8. Looping in @rickybanister and @joanrho to double check the correct color.

**Comment by glendaviesnz** [comment]
@olesyabrk ok, I see that the default wpcom guide overrides the default calypso guide blue, so I guess we need to do the same, will update it with the blue you had

**Comment by joanrho** [comment]
@olesyabrk Sorry if it wasn't clear when I sent over the AE file in Slack, but the background color of the single GIF we used in our more updated tour is our brand Blue 40/#3582c4 (see [Color Studio](https://color-studio.blog/)). 

For context, those updates were made as part of our pre-launch work, so it existed in another Figma file (Create Design > Pre-launch). Just updated that Welcome Tour Figma with page "Contextual Guidance i4" since it's the latest direction of the Welcome Tour (hasn't been implemented yet).

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4075986

Hi @glendaviesnz, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `inserter`
- `Meet the Classic block`
- `The block editor is now the default editor for all your sites, but you can still use the Classic block, if you prefer.`
- `Add the Classic block`
- `To use the classic block, click on the <InserterIconImage /> button at the top of the screen, search for Classic and insert the block.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4075986).

**Pull request merged**